src/targets/__tests__/maven.test.ts

```diff
@@ -303,3 +325,245 @@ describe('publish', () => {
     expect(cmdArgs[7]).toBe(DEFAULT_OPTION_VALUE);
   });
 });
+
+describe('closeAndReleaseRepository', () => {
+  test('should throw if repository is not opened', async () => {
+    const mvnTarget = createMavenTarget();
+    mvnTarget.getRepository = jest.fn(() =>
+      Promise.resolve(getRepositoryInfo('closed', false))
+    );
+    await expect(mvnTarget.closeAndReleaseRepository()).rejects.toThrow();
+  });
+
+  test('should call closeRepository and releaseRepository with fetched repository ID', async () => {
+    const mvnTarget = createMavenTarget();
+    mvnTarget.getRepository = jest.fn(() =>
+      Promise.resolve(getRepositoryInfo('open', false))
+    );
+    const callOrder: string[] = [];
+    mvnTarget.closeRepository = jest.fn(async () => {
+      callOrder.push('closeRepository');
+      return true;
+    });
+    mvnTarget.releaseRepository = jest.fn(async () => {
+      callOrder.push('releaseRepository');
+      return true;
+    });
+
+    await mvnTarget.closeAndReleaseRepository();
+
+    expect(mvnTarget.closeRepository).toHaveBeenCalledWith('sentry-java');
+    expect(mvnTarget.releaseRepository).toHaveBeenCalledWith('sentry-java');
+    expect(callOrder).toStrictEqual(['closeRepository', 'releaseRepository']);
+  });
+
+  test('should not release repostiory if it was not closed properly', async () => {
+    const mvnTarget = createMavenTarget();
+    mvnTarget.getRepository = jest.fn(() =>
+      Promise.resolve(getRepositoryInfo('open', false))
+    );
+    mvnTarget.closeRepository = jest.fn(() => Promise.reject());
+    mvnTarget.releaseRepository = jest.fn(() => Promise.resolve(true));
+
+    try {
+      await mvnTarget.closeAndReleaseRepository();
+    } catch (e) {
+      // no-empty
+    }
```

Comment:
Why not something like `expect(...).not.toThrow()`?

Reply:
Because we actually want to throw here

Reply:
Because imo assertions should match the name of the test.

> should not release repostiory if it was not closed properly

And assertion is that `releaseRepository` was never called. `closeAndReleaseRepository` is only a setup trigger.

Reply:
>Because imo assertions should match the name of the test

Yeah, that makes sense.

>And assertion is that releaseRepository was never called. closeAndReleaseRepository is only a setup trigger.

I think a comment here instead of `// on-empty` would go a long way to communicate this to other developers such as myself and @iker-barriocanal 🙂

Reply:
Fair point, will keep that in mind :)